### PR TITLE
Extend ParseIncomingWebhook function to handle tag related events

### PR DESCRIPTION
**Pull request by pavelmemory**
- [x] All [tests](https://github.com/jfrog/froggit-go/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `go fmt ./...` for formatting the code before submitting the pull request.
- [x] This feature is included on all supported VCS providers - GitHub, Bitbucket cloud, Bitbucket server, and GitLab.

---

The tag creation and removal webhook events now can be parsed by the `ParseIncomingWebhook()` function. The information about the tag is stored in the new `Tag` field of the `WebhookInfo` type. Also, the two new events added: `TagPushed` and `TagRemoved` to represent each of these operations.

### CODE REVIEW

File: vcsutils/webhookparser/testdata/github/tagcreatepayload.json [hunk 1]
```diff
@@ -0,0 +1 @@
...%22%3A%22https%3A%2F%2Fapi.github.com%2Fusers%2Fpavelmemory%2Freceived_events%22%2C%22type%22%3A%22User%22%2C%22site_admin%22%3Afalse%7D%2C%22created%22%3Atrue%2C%22deleted%22%3Afalse%2C%22forced%22%3Afalse%2C%22base_ref%22%3Anull%2C%22compare%22%3A%22https%3A%2F%2Fgithub.com%2Fpavelmemory%2Fgo-mockhttp%2Fcompare%2Ftag_intg%22%2C%22commits%22%3A%5B%5D%2C%22head_commit%22%3A%7B%22id%22%3A%225f1317136694845cceb328a0e30b232b9935b759%22%2C%22tree_id%22%3A%22bb26a84104023eab3977d0b6e4b0fa492ad9b002%22%2C%22distinct%22%3Atrue%2C%22message%22%3A%22Add+url+query+support+in+the+request+matcher%22%2C%22timestamp%22%3A%222019-10-24T22%3A51%3A16%2B03%3A00%22%2C%22url%22%3A%22https%3A%2F%2Fgithub.com%2Fpavelmemory%2Fgo-mockhttp%2Fcommit%2F5f1317136694845cceb328a0e30b232b9935b759%22%2C%22author%22%3A%7B%22name%22%3A%22Yinon+Avraham%22%2C%22email%22%3A%22yinonavraham%40gmail.com%22%2C%22username%22%3A%22yinonavraham%22%7D%2C%22committer%22%3A%7B%22name%22%3A%22Yinon+Avraham%22%2C%22email%22%3A%22yinonavraham%40gmail.com%22%2C%22username%22%3A%22yinonavraham%22%7D%2C%22added%22%3A%5B%5D%2C%22removed%22%3A%5B%5D%2C%22modified%22%3A%5B%22request_matcher.go%22%2C%22request_matcher_test.go%22%5D%7D%7D
```
Comment (cyrilc-pro) [inline vcsutils/webhookparser/testdata/github/tagcreatepayload.json:1]: Why is this not JSON?

Same for `vcsutils/webhookparser/testdata/github/tagdeletepayload.json`

Comment (pavelmemory) [reply vcsutils/webhookparser/testdata/github/tagcreatepayload.json:1]: The form-url encoded format is used from the beginning that is why I use it as well.

File: vcsutils/webhookparser/bitbucketcloud.go [hunk 2]
```diff
@@ -228,14 +266,27 @@ type bitbucketPush struct {
 }
 
 type bitbucketChange struct {
```
Comment (yahavi) [inline vcsutils/webhookparser/bitbucketcloud.go:268]: Looks like the "New" and the "Old" can be exported to a new struct to remove duplications.
I suggest doing the following:
1. Create a new struct "bitbucketResourceChange" (or something equivalent) with Name, Target, Type, and Message
2. Old and New will be of type bitbucketResourceChange

File: vcsutils/webhookparser/gitlab_test.go [hunk 3]
```diff
@@ -287,6 +287,78 @@ func TestGitLabParseIncomingPrWebhook(t *testing.T) {
 	}
 }
 
+func TestGitLab_parseIncomingWebhookTagEvents(t *testing.T) {
```
Comment (yahavi) [inline vcsutils/webhookparser/gitlab_test.go:290]:
```suggestion
func TestGitLabParseIncomingWebhookTagEvents(t *testing.T) {
```

File: vcsutils/webhookparser/webhookparser.go [hunk 4]
```diff
@@ -85,6 +87,22 @@ type WebHookInfoRepoDetails struct {
...
+	// Hash is a SHA of the tag.
+	Hash string `json:"hash"`
+	// TargetHash is a SHA of the commit the tag points to.
+	TargetHash string `json:"target_hash"`
+	// Message is a message used during tag creation if any.
+	Message string `json:"message"`
+	// Repository contains details about repository.
+	Repository WebHookInfoRepoDetails `json:"repository"`
+	// Author
+	Author WebHookInfoUser `json:"author,omitempty"`
```
Comment (yahavi) [inline vcsutils/webhookparser/webhookparser.go:103]: Let's add omitempty to all fields

**Pull request merged**